app/Platform/Controllers/HubController.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Controllers;
+
+use App\Data\UserPermissionsData;
+use App\Http\Controller;
+use App\Models\GameSet;
+use App\Models\System;
+use App\Models\User;
+use App\Platform\Actions\BuildGameListAction;
+use App\Platform\Actions\BuildGameSetRelatedHubsAction;
+use App\Platform\Actions\BuildHubBreadcrumbsAction;
+use App\Platform\Data\GameSetData;
+use App\Platform\Data\HubPagePropsData;
+use App\Platform\Data\SystemData;
+use App\Platform\Enums\GameListType;
+use App\Platform\Enums\GameSetType;
+use App\Platform\Requests\GameListRequest;
+use Illuminate\Http\RedirectResponse;
+use Inertia\Inertia;
+use Inertia\Response as InertiaResponse;
+use Jenssegers\Agent\Agent;
+
+// TODO mature hub needs to be age gated
+// TODO on rename hub, make sure it's always wrapped in square brackets
+// TODO play around on mobile
+
+class HubController extends Controller
+{
+    public function index(): void
+    {
+    }
+
+    public function store(): void
+    {
+    }
+
+    public function create(): void
+    {
+    }
+
+    public function show(GameListRequest $request, ?GameSet $gameSet): InertiaResponse|RedirectResponse
+    {
+        // On "/hubs", $gameSet will initially be null. We're trying to
+        // go to the central hub - manually find it and set the value.
+        if (!$gameSet->exists) {
+            $gameSet = GameSet::centralHub()->firstOrFail();
+        } else {
+            // If the user navigates directly to the central hub ID,
+            // we should redirect them to "/hubs" instead.
+            $centralHub = GameSet::centralHub()->first();
+            if ($centralHub && $gameSet->id === $centralHub->id) {
+                return redirect()->route('hub.index');
+            }
+        }
+
+        // Return a 404 if this game set isn't actually for a hub.
+        if ($gameSet->type !== GameSetType::Hub) {
```

Comment:
Consider use abort_if instead

Reply:
Good idea. This is from #2923, but I've made the change there and then remerged that branch into this one.